torch/multiprocessing/reductions.py

```diff
@@ -595,9 +597,10 @@ def reduce_typed_storage_child(storage):
 def reduce_storage(storage):
     from . import get_sharing_strategy
 
-    if storage.is_cuda:
+    current_device = torch._C._get_accelerator().type
+    if current_device != "cpu" and storage.device.type == current_device:
         raise RuntimeError(
-            "Cannot pickle CUDA storage; try pickling a CUDA tensor instead"
+            "Cannot pickle Device storage; try pickling a Device tensor instead"
```

Comment:
Better here would be to move this after the meta, and then just have this fail for everything that is not CPU. Looking at the code below, it doesn't look like it would work for non-CUDA non-CPU devices either

Reply:
@ezyang Modified